### PR TITLE
Validation for streamKey

### LINKED ISSUE #136: Spaces in bearer token isn't properly sanitized when loading on web
When putting a token into the web browser, all spaces are converted over to `%20` while what is received from OBS doesn't contain this sanitization. This causes a flow break when trying to load the stream, as you aren't allowed to send non-sanitized spaces in a web browser.
img

### PR DESCRIPTION
**Pull request by dilllxd**
_Should_ resolve #136, I've never really used Go so please check this over a couple times before merging. I tested locally and it seemed to work. Although, there could be easier/more preferred ways of validating the key.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -43,17 +44,36 @@ func logHTTPError(w http.ResponseWriter, err string, code int) {
 	http.Error(w, err, code)
 }
 
+func validateStreamKey(streamKey string) bool {
+	validKey := regexp.MustCompile(`^[a-zA-Z0-9_\-]+$`)
```
Comment (wolfcomp) [inline main.go:48]: This regex is missing 2 characters, both `~` and `.` are valid for HTTP

Comment (dilllxd) [reply main.go]: Thank you @wolfcomp, can you double check the new commit and make sure it's fixed?

**Comment by Sean-Der** [comment]
amazing! thank you so much @dilllxd 

Sorry I didn't respond sooner, was traveling for CommCon

**Comment by Sean-Der** [comment]
@dilllxd would you be ok if I added you to the repo? Would love to have you continue be involved

**Pull request merged**